### PR TITLE
Buildkite integration tests: Windows 2022 Sudo tests

### PR DESCRIPTION
**Pull request by pazone**
## What does this PR do?
Updates the new buildkite integration testing pipeline
Adds a new custom VM image `platform-ingest-elastic-agent-windows-2022` with pre-installed tools
Adds Windows 2022 integration tests.

## Details

Windows desktop images have not yet landed. Hence I moved golang installation from `pre-command.ps1` to a dedicated `install-go.ps1`. This script has to be called in the steps that don't use custom platform-ingest VM images. 

[Test build example](https://buildkite.com/elastic/elastic-agent-extended-testing-bk/builds/84#_)

## Why is it important?
We run the same tests as we do in OGC tests.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Related issues

- Relates https://github.com/elastic/ingest-dev/issues/4218

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1,3 +1,3 @@
 mage 1.14.0
-golang 1.22.8
```
Comment (swiatekm) [inline .tool-versions:2]: Do we need to make this change in this PR? Seems unrelated.

**Review by swiatekm — COMMENTED** [review]
Looks correct to me, but there's a bunch of changes included in this PR that seem unrelated. It would be easier to review if it only added the new integration test steps.

**Comment by pazone** [comment]
> Looks correct to me, but there's a bunch of changes included in this PR that seem unrelated. It would be easier to review if it only added the new integration test steps.

Unfortunately, all the changes are interconnected. We use the custom image with preinstalled tools - we have to skip go installation in the pre-command hook

**Comment by swiatekm** [comment]
> > Looks correct to me, but there's a bunch of changes included in this PR that seem unrelated. It would be easier to review if it only added the new integration test steps.
> 
> Unfortunately, all the changes are interconnected.
> 
>     * We use the new custom image with preinstalled golang -> we have to skip go installation in the pre-command hook
> 
>     * The pre-command hook changes affects windows unit tests and that's why their steps are updated

How will we update the preinstalled Golang version after merging this change? Right now, it can be updated without making any changes outside the agent repo.

**Comment by pazone** [comment]
> How will we update the preinstalled Golang version after merging this change? Right now, it can be updated without making any changes outside the agent repo. 
Our images are being rebuilt by cron. If a new golang version is not preinstalled we have a fallback option `asdf install` that installs it.
The `.tool-versions` changes are not very necessary here. The automatic version bump in `.tool-versions` is [planned](https://github.com/elastic/ingest-dev/issues/4302) for the next iteration.

**Review by swiatekm — APPROVED** [review]

**Review by oakrizan — APPROVED** [review]
lgtm
